v2/api/search/v1api20220901/structure.txt

```diff
@@ -0,0 +1,222 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/search/v1api20220901
+├── APIVersion: Enum (1 value)
+│   └── "2022-09-01"
+├── SearchService: Resource
+│   ├── Owner: github.com/Azure/azure-service-operator/v2/api/resources/v1apiv20191001.ResourceGroup
+│   ├── Spec: Object (14 properties)
+│   │   ├── AuthOptions: *Object (2 properties)
+│   │   │   ├── AadOrApiKey: *Object (1 property)
+│   │   │   │   └── AadAuthFailureMode: *Enum (2 values)
+│   │   │   │       ├── "http401WithBearerChallenge"
+│   │   │   │       └── "http403"
+│   │   │   └── ApiKeyOnly: map[string]v1.JSON
+│   │   ├── AzureName: string
+│   │   ├── DisableLocalAuth: *bool
+│   │   ├── EncryptionWithCmk: *Object (1 property)
+│   │   │   └── Enforcement: *Enum (3 values)
+│   │   │       ├── "Disabled"
+│   │   │       ├── "Enabled"
+│   │   │       └── "Unspecified"
+│   │   ├── HostingMode: *Enum (2 values)
+│   │   │   ├── "default"
+│   │   │   └── "highDensity"
+│   │   ├── Identity: *Object (1 property)
+│   │   │   └── Type: *Enum (2 values)
+│   │   │       ├── "None"
+│   │   │       └── "SystemAssigned"
+│   │   ├── Location: *string
+│   │   ├── NetworkRuleSet: *Object (1 property)
+│   │   │   └── IpRules: Object (1 property)[]
+│   │   │       └── Value: *string
```

Comment:
Wait is this supposed to be an ARM ID? We may need to annotate it as such in our YAML

Reply:
Not really, have a look at the [docs](https://learn.microsoft.com/en-us/azure/templates/microsoft.search/2021-04-01-preview/searchservices?pivots=deployment-language-bicep#iprule)

Reply:
Looks good.